packages/core/content-manager/server/src/history/services/history.ts

```diff
@@ -105,9 +106,14 @@ const createHistoryService = ({ strapi }: { strapi: Core.Strapi }) => {
                       return null;
                     }
 
-                    return strapi
-                      .query('admin::user')
-                      .findOne({ where: { id: userToPopulate.id } });
+                    return strapi.query('admin::user').findOne({
+                      where: {
+                        ...(userToPopulate.id ? { id: userToPopulate.id } : {}),
+                        ...(userToPopulate.documentId
+                          ? { documentId: userToPopulate.documentId }
+                          : {}),
+                      },
+                    });
```

Comment:
@alexandrebodin I don't mind the change but I need some context:
- what's the bug you were seeing and how can I test it?
- are admin users supposed to just have id, or just document id, or both? Why is it not the same every time?

Reply:
The history code is buggy. It's only storing in db the documentId, locale and published information and not the id. So When this code is executed (create a relation to an admin user in a content type to test it) it doesn't have any id in the userToPopulate and just breaks. 

I'm just adding documentId and not removing id in case some data was saved with an id previsouly. I don't know when this was broken

Reply:
Discussed live. UI fix incoming from Rémi 👍